haskell/private/cc_wrapper.bzl

```diff
@@ -0,0 +1,56 @@
+load("@bazel_tools//tools/cpp:toolchain_utils.bzl", "find_cpp_toolchain")
+load("@bazel_tools//tools/build_defs/cc:action_names.bzl", "ACTION_NAMES")
+load("@os_info//:os_info.bzl", "is_linux")
+
+def _cc_wrapper_impl(ctx):
+    cc_toolchain = find_cpp_toolchain(ctx)
+    feature_configuration = cc_common.configure_features(
+        ctx = ctx,
+        cc_toolchain = cc_toolchain,
+        requested_features = ctx.features,
+        unsupported_features = ctx.disabled_features,
+    )
+    cc = cc_common.get_tool_for_action(
+        feature_configuration = feature_configuration,
+        action_name = ACTION_NAMES.c_compile,
+    )
```

Comment:
[`CcToolchainInfo.compiler_executable`](https://docs.bazel.build/versions/0.28.0/skylark/lib/CcToolchainInfo.html#compiler_executable) has been deprecated. `get_tool_for_action` is used instead to get access to the compiler executable. See also https://github.com/bazelbuild/bazel/issues/8802. This will return the C compiler of the active cc_toolchain.